src/components/workflow/workflow.service.ts

```diff
@@ -0,0 +1,866 @@
+/* eslint-disable */
+import { Injectable, NotImplementedException, NotFoundException } from '@nestjs/common';
+import { generate } from 'shortid';
+import { node, relation } from 'cypher-query-builder';
+import { ISession } from '../../common';
+import { CreateWorkflow, Workflow, AddState, State, UpdateState, StateListOutput, GroupState, ChangeState, PossibleState, RequiredField, CommentState, RequiredFieldListOutput, FiledObject } from './dto';
+import { Logger, ILogger, DatabaseService, matchSession } from '../../core';
+
+@Injectable()
+export class WorkflowService {
+  constructor(
+    private readonly db: DatabaseService,
+    @Logger('workflow.service') private readonly logger: ILogger
+  ) {}
+
+  async readOneState(session: ISession, stateId: string): Promise<State> {
+    const result = await this.db
+      .query()
+      .match([
+        node('token', 'Token', {
+          active: true,
+          value: session.token
+        }),
+        relation('in', '', 'token', {
+          active: true
+        }),
+        node('user'),
+        relation('in', '', 'admin', {
+          active: true
+        }),
+        node('baseNode'),
+        relation('out', '', 'workflow', {
+          active: true
+        }),
+        node('workflow'),
+        relation('out', '', 'possibleState', {
+          active: true
+        }),
+        node('state', 'State', {
+          id: stateId
+        })
+      ])
+      .return([
+        {
+          state: [
+            {id: 'id'},
+            {value: 'value'}
+          ]
+        }
+      ])
+      .first();
+
+    if( !result ) {
+      throw new NotFoundException('could not find a state');
+    }
+
+    return {
+      id: result.id,
+      value: result.value
+    }
+  }
+  // only 1 workflow per base node, only 1 base node per workflow. WF doesn't need a name, just id
+  // when the workflow node is created, the :CurrentState node is created.
```

Comment:
Let's delete this comment. We'll eventually enable multiple workflows on a node.